packages/@aws-cdk/aws-secretsmanager/lib/secret.ts

```diff
@@ -129,13 +129,6 @@ export interface SecretAttributes {
    * The ARN of the secret in SecretsManager.
    */
   readonly secretArn: string;
-
-  /**
-   * The name of the secret in SecretsManager.
-   *
-   * @default - the name is derived from the secretArn.
-   */
-  readonly secretName?: string;
```

Comment:
I would argue that a better DX is to only support `secretName`, no?

Reply:
> to only support secretName

We can't stop supporting `secretArn` as an attribute, no.

You mean make `secretArn` optional, and add back in `secretName` as optional, then a validation in `fromSecretAttributes` that at least one is present? We _can_ do that.

Reply:
> > to only support secretName
> 
> 
> 
> We can't stop supporting `secretArn` as an attribute, no.
> 
> 
> 
> You mean make `secretArn` optional, and add back in `secretName` as optional, then a validation in `fromSecretAttributes` that at least one is present? We _can_ do that.

I guess that could work although generally against our api guidelines (normally in such cases we prefer something like union-like classes). I am okay with how it is now